### PR TITLE
Extra panels to create yaml

### PR DESCRIPTION
**Pull request by Jakob37**
# Description and reviewer info

Add support for including extra panels to institutes beyond those included in the institute itself.

These panels are configured through a JSON with institute key -> additional panel IDs.

The extra panels JSON is currently versioned in the Scout infrastructure repo (https://github.com/SMD-Bioinformatics-Lund/Scout-infrastructure).

Some refactor in `create_yaml.pl`. Seems to work well during testing, but an extra pair of eyes is appreciated.

## Type of change

- [ ] Documentation
- [ ] Patch
- [x] Minor change
- [ ] Major change 

# Checklist

- [x] Self-review of my code
- [x] Update the CHANGELOG
- [x] Tag the latest commit (vX.Y.Z format)

## Documentation
- [ ] At least one other person has reviewed my changes (not required for trivial changes)

## Patch
- [ ] Stub run completes without errors or new warnings
- [ ] At least one other person has reviewed and approved my code (not required for trivial changes)

## Major / Minor change
- [x] Stub run completes without errors or new warnings
- [x] GIAB single (wgs) finishes and differences to current master branch have been investigated (using [PipeEval](https://github.com/Clinical-Genomics-Lund/PipeEval))
- [x] GIAB trio (wgs) finishes and differences to current master branch have been investigated
- [x] Seracare (onco) sample finishes and differences to current master branch have been investigated
- [x] Output from processes with altered code have been inspected (i.e. in the work folder: .command.sh, .command.log, .command.err and result files)
- [x] All three samples can be loaded into Scout
- [x] At least one other person has reviewed and approved my code
- [x] I have made corresponding changes to the documentation (software versions, etc.)

# Test/review documentation

## Review performed by

- [ ] Alexander
- [ ] Jakob
- [ ] Paul
- [ ] Ryan
- [x] Viktor

(Add if missing)

## Testing performed by

- [ ] Alexander
- [x] Jakob
- [ ] Paul
- [ ] Ryan
- [ ] Viktor

### CODE REVIEW

**Comment by Jakob37** [comment]
New somatic panels can be seen on wgs case loaded into Scout:

new_cancer_panels

File: bin/create_yml.pl [hunk 1]
```diff
@@ -1,13 +1,39 @@
...
+    'genome=s',
+    'antype=s',
+    'ped=s',
+    'assay=s',
+    'files=s',
+    'panelsdef=s',
+    'extra_panels=s'
+) or usage();
+
+my @required = ('g', 'd', 'out', 'files', 'ped', 'panelsdef', 'assay', 'antype');
```
Comment (ViktorHy) [inline bin/create_yml.pl:26]: good addition

File: configs/nextflow.hopper.config [hunk 2]
```diff
@@ -69,6 +69,7 @@ params {
   // SCOUT //
   scoutbed          = "${refpath}/bed/scout/hgnc.grch38p13.exons.sort.bed"
   panelsdef         = "/fs1/resources/scout/scout_active_gene_panels_LATEST.json"
+  extra_panels      = "/fs1/resources/scout/scout_infrastructure/wgs_conf/extra_panels.json"
```
Comment (ViktorHy) [inline configs/nextflow.hopper.config:72]: good enough solution I guess,

File: main.nf [hunk 3]
```diff
@@ -4040,7 +4040,15 @@ process create_yaml {
...
-			--g $group,$clarity_sample_id --d $diagnosis --panelsdef $params.panelsdef --out ${group}.yaml --ped $ped --files $INFO --assay $assay,$analysis --antype $params.antype
+			--g $group,$clarity_sample_id \\
+			--d $diagnosis \\
+			--panelsdef $params.panelsdef \\
+			--out ${group}.yaml \\
+			--ped $ped \\
+			--files $INFO \\
+			--assay $assay,$analysis \\
+			--antype $params.antype \\
+			--extra_panels $params.extra_panels
```
Comment (ViktorHy) [inline main.nf:4051]: I guess this also works for onco? can you give an empty --extra_panels?

**Review by ViktorHy — COMMENTED** [review]
Look like a good solution to our problem. Nice touchup of code as well. I would just like to see that profiles without extra panels also works.

**Comment by ViktorHy** [comment]
I'm good then. Go for merge when you feel confident!

**Pull request merged**